zingolib/src/lightclient/describe.rs

```diff
@@ -375,104 +396,113 @@ impl LightClient {
     pub fn get_server_uri(&self) -> http::Uri {
         self.config.get_lightwalletd_uri()
     }
-
-    fn tx_summary_matcher(
+    // Given a transaction write down ValueTransfer information in a Summary list
+    fn record_value_transfers(
         summaries: &mut Vec<ValueTransfer>,
         txid: TxId,
-        transaction_md: &TransactionRecord,
-    ) {
+        transaction_record: &TransactionRecord,
+        transaction_records: &TransactionRecordsById,
+    ) -> Result<(), ValueTransferRecordingError> {
+        let is_received = match transaction_records.transaction_is_received(transaction_record) {
+            Ok(received) => received,
+            Err(fee_error) => {
+                return Err(ValueTransferRecordingError::FeeCalculationError(
+                    fee_error.to_string(),
+                ))
+            }
+        };
```

Comment:
let-else statements are really nice to tidy up this `Ok(r) => r` fluff
```suggestion
        let Ok(is_received) = transaction_records.transaction_is_received(transaction_record) else {
                return Err(ValueTransferRecordingError::FeeCalculationError(
                    fee_error.to_string()
                ))
        };
```